packages/devtools/lib/src/debugger/debugger.dart

```diff
@@ -428,9 +478,43 @@ class DebuggerScreen extends Screen {
         ..click(() => breakpointsView.element.toggleAttribute('hidden')),
       breakpointsView.element,
       PNavMenuItem('Scripts')
-        ..add(
-          scriptCountDiv = span(text: '-', c: 'counter'),
-        )
+        ..add([
+          textfield
+            ..click(() {
+              _matcher ??=
+                  ScriptsMatcher(scriptsView, textfield, debuggerState);
+              scriptsView.setMatcher(_matcher);
+            })
+            ..focus(() {
+              _matcher ??=
+                  ScriptsMatcher(scriptsView, textfield, debuggerState);
+              scriptsView.setMatcher(_matcher);
+            })
+            ..onKeyUp.listen((html.KeyboardEvent e) {
+              switch (e.keyCode) {
+                case DOM_VK_RETURN:
+                case DOM_VK_ESCAPE:
+                case DOM_VK_PAGE_UP:
+                case DOM_VK_PAGE_DOWN:
+                case DOM_VK_END:
+                case DOM_VK_HOME:
+                case DOM_VK_UP:
+                case DOM_VK_DOWN:
+                  return;
+                default:
+                  final html.InputElement inputElement = textfield.element;
+                  final String value = inputElement.value.trim();
+
+                  if (!_matcher.active) {
+                    _matcher.start();
+                  }
+                  _matcher.displayMatchingScripts(value);
+              }
+            })
+            ..onCut.listen((html.ClipboardEvent e) => e.preventDefault())
```

Comment:
comment why you need to prevent the default action for cut and paste

Reply:
This was something I was toying with not needed - removed.